src/dataDisplay/table.jsx

```diff
@@ -1,21 +1,522 @@
+import {
+    DEBOUNCE_WAIT_TIME,
+    TH_RESIZABLE_MAX_WIDTH,
+} from './tableConstants';
 import TableHeader from './tableHeader';
-import TableHeaderCell from './tableHeaderCell';
 import TableRow from './tableRow';
+import TableHeaderCell from './tableHeaderCell';
+
+const propTypes = {
+    /**
+     * Basic (default) styling.
+     */
+    basic: PropTypes.bool,
+    /**
+     * If `true`, all columns will be divided.
+     */
+    celled: PropTypes.bool,
+    /**
+     * The content of the Table.
+     */
+    children: PropTypes.node,
+    /**
+     * Override or extend the styles applied to Table.
+     */
+    classes: PropTypes.shape({
+        tableSticky: PropTypes.string,
+        tableStickyColumns: PropTypes.string,
+    }),
+    /**
+     * Add additional classes to the Table.
+     */
+    className: PropTypes.string,
+    /**
+     * If `true`, applies `width: auto` to the Table.
+     */
+    collapsing: PropTypes.bool,
+    /**
+     * If `true`, far left column divided from the rest of the columns.
+     */
+    definition: PropTypes.bool,
+    /**
+     * If `true`, columns can be evenly spaced.
+     */
+    fixed: PropTypes.bool,
+    /**
+     * Change the font size of typography in the Table.
+     */
+    fontSize: PropTypes.oneOf(Utils.sizeEnums()),
+    /**
+     * Deprecated prop.
+     */
+    fullWidth: PropTypes.bool,
+    /**
+     * Add an id to the Table.
+     */
+    id: PropTypes.string,
+    /**
+     * Table row and cells can be highlighted as if they have been selected.
+     */
+    selectable: PropTypes.bool,
+    /**
+     * Truncate copy within cells in the Table
+     */
+    singleLine: PropTypes.bool,
+    /**
+     * Change the cell's vertical size in the Table.
+     */
+    size: PropTypes.oneOf(['l', 'large', 'm', 'medium', 's', 'small']),
+    /**
+     * Deprecated prop.
+     */
+    stackable: PropTypes.bool,
+    /**
+     * A table can have one or more columns defined as 'sticky' so that they stay fixed while the
+     * user horizontally scrolls to see the remaining columns.
+     */
+    stickyColumnCount: PropTypes.number,
+    /**
+     * Resizable cell default percentage width related to the Table container size.
+     */
+    resizableColumnWidthPercentage: (props, propName, componentName) => {
+        const {
+            [propName]: propValue,
+        } = props;
+        const isValid = propValue ?
+            propValue > 0 && propValue <= parseInt(TH_RESIZABLE_MAX_WIDTH * 100, 10) :
+            true;
+        if (!isValid) {
+            return new Error(
+                `Invalid prop value for ${propName} (${propValue}) supplied to ${componentName}. Validation failed.`,
+            );
+        }
+
+        return null;
+    },
+    /**
+     * If `true` or `very`, the Table will bleed off the edge.
+     */
+    stretch: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(['very']),
+    ]),
+    /**
+     * Deprecated prop.
```

Comment:
Same: Any additional information about why this is deprecated or what replaces it, if anything?

Reply:
We don't support it anymore per a conversation with @aaronsampson.

Reply:
Resolving.